tensorboard/uploader/exporter_test.py

```diff
@@ -199,6 +200,61 @@ def stream_experiments(request, **kwargs):
     self.assertIn("../authorized_keys", msg)
     mock_api_client.StreamExperimentData.assert_not_called()
 
+  def test_fails_nicely_on_stream_experiment_data_timeout(self):
+    # Setup: Client where:
+    #   1. stream_experiments will say there is one experiment_id.
+    #   2. stream_experiment_data will raise a grpc CANCELLED, as per
+    #      a timeout.
+    mock_api_client = self._create_mock_api_client()
+    experiment_id="123"
+
+    def stream_experiments(request, **kwargs):
+      del request  # unused
+      yield export_service_pb2.StreamExperimentsResponse(
+          experiment_ids=[experiment_id])
+
+    def stream_experiment_data(request, **kwargs):
+      raise test_util.grpc_error(grpc.StatusCode.CANCELLED, "details string")
+
+    mock_api_client.StreamExperiments = stream_experiments
+    mock_api_client.StreamExperimentData = stream_experiment_data
+
+    outdir = os.path.join(self.get_temp_dir(), "outdir")
+    # Execute: exporter.export()
+    exporter = exporter_lib.TensorBoardExporter(mock_api_client, outdir)
+    generator = exporter.export()
+    # Expect: A nice exception of the right type and carrying the right
+    #   experiment_id.
+    with self.assertRaises(exporter_lib.GrpcTimeoutException) as cm:
+      next(generator)
+    self.assertEquals(cm.exception.experiment_id, experiment_id)
+
+  def test_stream_experiment_data_passes_through_unexpected_exception(self):
```

Comment:
> <code>def test_<b>stream_experiment_data</b>_passes_through_unexpected_exception</code>

nit: You’re testing properties about the exporter, not the internal API
method `StreamExperimentData`. Maybe you meant

```
test_passes_through_unexpected_stream_experiment_data_exception
```

?

Reply:
Done. (tested)
$ bazel run :tensorboard -- dev --endpoint=dns:///api-staging.tensorboard.dev:443 --grpc_creds_type=ssl export --outdir=/tmp/export_020 --alsologtostderr
INFO: Analyzed target //tensorboard:tensorboard (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //tensorboard:tensorboard up-to-date:
  bazel-bin/tensorboard/tensorboard
INFO: Elapsed time: 1.076s, Critical Path: 0.10s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
INFO: Running command line: bazel-bin/tensorboard/tensorboard dev '--endpoint=dns:///api-staging.tensorboard.dev:443' '--grpc_creds_type=ssl' export '--outdir=/tmp/export_020' --INFO: Build completed successfully, 1 total action
WARNING:root:Limited tf.compat.v2.summary API due to missing TensorBoard installation.
WARNING:root:Limited tf.compat.v2.summary API due to missing TensorBoard installation.
WARNING:root:Limited tf.compat.v2.summary API due to missing TensorBoard installation.
WARNING:root:Limited tf.compat.v2.summary API due to missing TensorBoard installation.
WARNING:root:Limited tf.compat.v2.summary API due to missing TensorBoard installation.
WARNING:root:Limited tf.compat.v2.summary API due to missing TensorBoard installation.
WARNING:root:Limited tf.summary API due to missing TensorBoard installation.
Downloaded experiment P3Ek7gUAQrGDaPmwSgpeSQ

Uploader has failed because of a timeout error.  Please reach out via e-mail to tensorboard.dev-support@google.com to get help completing your export of experiment v8Dn7OvCRI2K25n4CaYjBA.
Done. Downloaded 1 experiments to: /tmp/export_020